src/commitments.rs

```diff
@@ -39,8 +39,8 @@ impl MultiCommitGens {
     }
   }
 
-  pub fn split_at_mut(&mut self, mid: usize) -> (MultiCommitGens, MultiCommitGens) {
-    let (G1, G2) = self.G.split_at_mut(mid);
```

Comment:
(Either `split_at` or `split_at_mut` works fine since you copy the two slices, but I think the former is preferable since it doesn't require a mutable borrow.)

Reply:
Sounds good!